pepsico/projections/maproom.py

```diff
@@ -176,6 +212,59 @@ def plot_ts(ts, name, color, start_format, units):
         )
 
 
+    @APP.callback(
+        Output("btn_csv", "disabled"),
+        Input("lat_input", "value"),
+        Input("lng_input", "value"),
+        Input("lat_input", "min"),
+        Input("lng_input", "min"),
+        Input("lat_input", "max"),
+        Input("lng_input", "max"),
+    )
+    def invalid_button(lat, lng, lat_min, lng_min, lat_max, lng_max):
+        return (
+            lat < float(lat_min) or lat > float(lat_max)
+            or lng < float(lng_min) or lng > float(lng_max)
+        )
+
+
+    @APP.callback(
+        Output("download-dataframe-csv", "data"),
+        Input("btn_csv", "n_clicks"),
+        State("loc_marker", "position"),
+        State("region", "value"),
+        State("model", "value"),
+        State("variable", "value"),
+        State("start_month", "value"),
+        State("end_month", "value"),
+        prevent_initial_call=True,
+    )
+    def send_data_as_csv(
+        n_clicks, marker_pos, region, model, variable, start_month, end_month,
+    ):
+        lat = marker_pos[0]
+        lng = marker_pos[1]
+        start_month = ac.strftimeb2int(start_month)
+        end_month = ac.strftimeb2int(end_month)
+        data_ds, error_msg = local_data(
+            lat, lng, region, model, variable, start_month, end_month
+        )
+        if error_msg == None :
+            lng_units = "E" if (lng >= 0) else "W"
+            lat_units = "N" if (lat >= 0) else "S"
+            file_name = (
+                f'{data_ds["histo"]["T"].dt.strftime("%b")[0].values}-'
+                f'{data_ds["histo"]["seasons_ends"].dt.strftime("%b")[0].values}'
+                f'_{variable}_{model}_{abs(lat)}{lat_units}_{abs(lng)}{lng_units}'
+                f'.csv'
+            )
+            df = data_ds.to_dataframe()
+        else :
+            file_name = "nodata.csv"
+            df = pd.DataFrame()
```

Comment:
Seems like it would be more appropriate to display an error message rather than downloading a file that we know will be empty. (Not a big deal.)

Reply:
I didn't realize I could return None. That is done. As for the error message: it's printed on the empty graph, and the Download button is disabled, so we should be good here.